### PR TITLE
[SELC-3924] feat: Added maven settings to set token and access to registry

### PR DESCRIPTION
**Pull request by pierpaolodidato89**
#### List of Changes

Added maven settings to set token and access to registry

### CODE REVIEW

**Review by andrea-putzu — APPROVED** [review]

**Review by Krusty93 — CHANGES_REQUESTED** [review]
Creating a dependency between the environment and the Dockerfile is considered a bad practice. A solution for this would be accepting values for `github.actor` and `secrets.GITHUB_TOKEN` through the use of build args. So, both the new step `Maven settings` and `COPY ./.m2/settings.xml` can be removed

**Review by andrea-putzu — APPROVED** [review]

File: apps/user-cdc/Dockerfile [hunk 1]
```diff
@@ -14,8 +14,24 @@ WORKDIR /src/apps/user-cdc
...
-WORKDIR /src
-RUN mvn --projects :user-cdc --also-make clean package -DskipTests
+RUN echo "<settings>\n" \
+         "<servers>\n" \
+		 "<server>\n" \
+		 "<id>\${repositoryId}</id>\n" \
+		 "<username>\${repoLogin}</username>\n" \
+		 "<password>\${repoPwd}</password>\n" \
+		 "</server>\n" \
+		 "</servers>\n" \
```
Comment (Krusty93) [inline apps/user-cdc/Dockerfile:24]: Can you please use spaces instead of tabs here?

File: apps/user-cdc/Dockerfile [hunk 2]
```diff
@@ -14,8 +14,24 @@ WORKDIR /src/apps/user-cdc
...
+         "</settings>\n" > settings.xml
+
+RUN cat settings.xml
```
Comment (Krusty93) [inline apps/user-cdc/Dockerfile:27]: Can be removed

File: apps/user-ms/Dockerfile [hunk 3]
```diff
@@ -14,8 +14,24 @@ WORKDIR /src/apps/user-ms
...
+         "<servers>\n" \
+		 "<server>\n" \
+		 "<id>\${repositoryId}</id>\n" \
+		 "<username>\${repoLogin}</username>\n" \
+		 "<password>\${repoPwd}</password>\n" \
+		 "</server>\n" \
+		 "</servers>\n" \
+         "</settings>\n" > settings.xml
+
+RUN cat settings.xml
```
Comment (Krusty93) [inline apps/user-ms/Dockerfile:27]: Can be removed

File: apps/user-cdc/Dockerfile [hunk 4]
```diff
@@ -14,8 +14,24 @@ WORKDIR /src/apps/user-cdc
...
+
+ARG REPO_ID
+ARG REPO_USERNAME
+ARG REPO_PASSWORD
+ENV REPO_ID=${REPO_ID} REPO_USERNAME=${REPO_USERNAME} REPO_PASSWORD=${REPO_PASSWORD}
```
Comment (Krusty93) [inline apps/user-cdc/Dockerfile:32]: I'd remove the ENV to use ARG in the line below, what you think?

File: apps/user-ms/Dockerfile [hunk 5]
```diff
@@ -14,8 +14,24 @@ WORKDIR /src/apps/user-ms
...
+
+ARG REPO_ID
+ARG REPO_USERNAME
+ARG REPO_PASSWORD
+ENV REPO_ID=${REPO_ID} REPO_USERNAME=${REPO_USERNAME} REPO_PASSWORD=${REPO_PASSWORD}
```
Comment (Krusty93) [inline apps/user-ms/Dockerfile:32]: I'd remove the ENV to use ARG in the line below, what you think?

File: infra/container_apps/user-cdc/env/dev/terraform.tfvars [hunk 6]
```diff
@@ -1,6 +1,5 @@
 prefix    = "selc"
```
Comment (Krusty93) [inline infra/container_apps/user-cdc/env/dev/terraform.tfvars:1]: Can be removed, added by module

File: infra/container_apps/user-ms/env/dev/terraform.tfvars [hunk 7]
```diff
@@ -1,6 +1,5 @@
 prefix    = "selc"
```
Comment (Krusty93) [inline infra/container_apps/user-ms/env/dev/terraform.tfvars:1]: Can be removed, added by module

**Review by Krusty93 — APPROVED** [review]
Few suggestions

**Review by Krusty93 — APPROVED** [review]

**Pull request merged**